x-pack/plugins/ml/public/application/datavisualizer/data_drift/data_view_editor.tsx

```diff
@@ -0,0 +1,190 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import useDebounce from 'react-use/lib/useDebounce';
+import useObservable from 'react-use/lib/useObservable';
+import React, { ChangeEvent, ReactNode, useMemo, useState, useEffect } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n-react';
+
+import {
+  EuiBasicTable,
+  EuiCallOut,
+  EuiFieldText,
+  EuiFlexItem,
+  EuiFormRow,
+  EuiFlexGrid,
+  useEuiTheme,
+} from '@elastic/eui';
+import type { DataViewEditorService } from '@kbn/data-view-editor-plugin/public';
+import type { MatchedItem } from '@kbn/data-views-plugin/public';
+import { useTableSettings } from '../../data_frame_analytics/pages/analytics_management/components/analytics_list/use_table_settings';
+import { canAppendWildcard, matchedIndicesDefault } from './data_drift_index_patterns_editor';
+
+interface DataViewEditorProps {
+  label: ReactNode;
+  dataViewEditorService: DataViewEditorService;
+  indexPattern: string;
+  setIndexPattern: (ip: string) => void;
+  onError: (errorMsg?: string) => void;
+  helpText?: ReactNode;
+}
+
+const mustMatchError = i18n.translate(
+  'xpack.ml.dataDrift.indexPatternsEditor.createIndex.noMatch',
+  {
+    defaultMessage: 'Name must match one or more data streams, indices, or index aliases.',
+  }
+);
+
+export function DataViewEditor({
+  label,
+  dataViewEditorService,
+  indexPattern,
+  setIndexPattern,
+  onError,
+  helpText,
+}: DataViewEditorProps) {
+  useDebounce(
+    () => {
+      dataViewEditorService.setIndexPattern(indexPattern);
+    },
+    250,
```

Comment:
Because as the input changes, dataViewEditorService is also updating the list of time fields available & validating if the same name already exists. No need to make this call every key stroke change.